File: 7_kyu/Wealth_equality_finally.lua
local solution = {}

-- Return arithmetic mean value
function mean(x)
  if not x or #x == 0 then
    return nil
  end
  local sum = 0
  for _,v in pairs(x) do
    sum = sum + v
  end
  return sum/#x
end

-- Set all items to the mean value.
function solution.redistribute_wealth(wealth)
  if wealth and #wealth > 0 then
    local mean = mean(wealth)
    for k,_ in pairs(wealth) do
      wealth[k] = mean
    end
  end
end

return solution

File: 7_kyu/Wealth_equality_finally.py
# Replaces all items in the list by the mean value.
def redistribute_wealth(wealth: list) -> None:
    if not wealth or len(wealth) < 1:
        # Nothing to do
        return
    mean = sum(wealth)/len(wealth)
    for i in range(len(wealth)):
        wealth[i] = mean